src/sink_tests.cc

```diff
@@ -410,6 +410,145 @@ TEST_F(SinkTest, RandomNormalWithMeanSttdev) {
   EXPECT_NEAR(qr.GetVal<double>("Quantity"), 1.52979, 0.0001);
 }
 
+TEST_F(SinkTest, RandomUniformFreq) {
+  using cyclus::QueryResult;
+  using cyclus::Cond;
+
+  std::string config =
+    "   <in_commods>"
+    "     <val>commods_1</val>"
+    "   </in_commods>"
+    "   <capacity>10</capacity>"
+    "   <random_frequency_type>UniformInt</random_frequency_type> "
+    "   <random_frequency_min>2</random_frequency_min> "
+    "   <random_frequency_max>4</random_frequency_max> ";
+
+  int simdur = 3;
+  cyclus::MockSim sim(cyclus::AgentSpec
+          (":cycamore:Sink"), config, simdur);
+  sim.AddSource("commods_1").capacity(10).Finalize();
+  int id = sim.Run();
+
+  QueryResult qr = sim.db().Query("Transactions", NULL);
+  EXPECT_EQ(qr.rows.size(), 1);
+  int trans_time = qr.GetVal<int>("Time", 0);
```

Comment:
Can you add a comment about what you are querying and why this is the right answer?